crates/compiler/mono/src/layout.rs

```diff
@@ -3460,6 +3467,7 @@ fn layout_from_flat_type<'a>(
         EmptyTagUnion => cacheable(Ok(Layout::VOID)),
         EmptyRecord => cacheable(Ok(Layout::UNIT)),
         EmptyTuple => cacheable(Ok(Layout::UNIT)),
+        EffectfulFunc => internal_error!(),
```

Comment:
minor: would be good to add a message describing the issue, even a couple words

Reply:
Done in https://github.com/roc-lang/roc/pull/7170/commits/6dbf5a39ef809e0b5a9209f401156e6b26287a8c